docs/Data/Injector.md

```diff
@@ -0,0 +1,52 @@
+## Module Data.Injector
+
+#### `Injector`
+
+``` purescript
+data Injector a b
+  = Injector (a -> b) (b -> Maybe a)
```

Comment:
Yeah, that's exactly what it is really - we don't have a dependency on any lens/optic stuff at the moment, but maybe we should just use that instead?